contrib/epee/include/epee/net/enums.h

```diff
@@ -55,10 +53,19 @@ namespace net_utils
 		tor = 3
 	};
 
-	// implementations in src/net_utils_base.cpp
+    //! \return String name of address type
+    constexpr std::string_view to_string(address_type a) noexcept {
+      switch (a) {
+        case address_type::ipv4: return "IPv4"sv;
+        case address_type::ipv6: return "IPv6"sv;
+        case address_type::i2p: return "I2P"sv;
+        case address_type::tor: return "Tor"sv;
+        default: return "invalid"sv;
+      }
+    }
 
 	//! \return String name of zone or "invalid" on error.
-    constexpr std::string_view zone_to_string(zone value) noexcept {
+    constexpr std::string_view to_string(zone value) noexcept {
```

Comment:
would this be better as a template specialization of `std::to_string` instead?

Reply:
rather and override?

Reply:
No, defining stuff in `std` is disallowed unless explicitly noted (e.g. std::hash).  `std::to_string` isn't.